### PR TITLE
CUMULUS-764 Integration tests should clean up after themselves

### PR DESCRIPTION
**Pull request by mennovandiermen**
**Summary:** Summary of changes

Addresses [CUMULUS-764: Integration tests should clean up after themselves](https://bugs.earthdata.nasa.gov/browse/CUMULUS-764)

## Changes

* New `timestampedTestDataFolder` function to create S3 prefix based on stackname, test and timestamp.
* New `uploadTestDataToBucket` function to upload test data at the start of the test.
* New `deleteFolder` function to clean up staged test data.
* S3 provider tests updated to use these functions in `beforeAll` and `afterAll` calls.
* New `apiTestUtils.deleteGranule` function to delete ingested test granules via API (API delete call has s3 file deletion logic).
* Fixes to `del` in `endpoints/granule.js` to not fail when encountering a file in the granule’s `files` list that isn’t on S3. (e.g. granules that failed when syncing file 2 of 3)
* Kinesis tests set up/teardown their own streams with `timestampedStreamName` functions.
* Update test inputs/outputs with test-created temp folders.
* Delete providers and collections after tests complete.

## Test Plan
Things that should succeed before merging.

- [x] Unit tests
- [x] Integration tests
- [x] Adhoc testing
- [x] update CHANGELOG

### CODE REVIEW

File: example/spec/setup/PopulateProvidersCollections.js [hunk 1]
```diff
@@ -2,9 +2,7 @@
 
 const path = require('path');
 const { addProviders, addCollections, addRules } = require('@cumulus/integration-tests');
-const { s3 } = require('@cumulus/common/aws');
-const { loadConfig } = require('../helpers/testUtils');
-const fs = require('fs-extra');
+const { loadConfig, uploadTestDataToBucket } = require('../helpers/testUtils');
```
Comment (laurenfrederick) [inline example/spec/setup/PopulateProvidersCollections.js]: Does this file have any references to `uploadTestDataToBucket`?

File: packages/integration-tests/api.js [hunk 2]
```diff
@@ -162,6 +162,26 @@ async function applyWorkflow({ prefix, granuleId, workflow }) {
   return JSON.parse(payload.body);
 }
 
+async function deleteGranule({ prefix, granuleId }) {
```
Comment (laurenfrederick) [inline packages/integration-tests/api.js]: This function needs to be documented.

File: example/spec/setup/PopulateProvidersCollections.js (hunk 1 above)
Comment (mennovandiermen) [reply example/spec/setup/PopulateProvidersCollections.js]: No, it used to. Will remove.

File: packages/integration-tests/api.js (hunk 2 above)
Comment (mennovandiermen) [reply packages/integration-tests/api.js]: Agreed.

File: example/spec/discoverAndQueuePdrs/DiscoverAndQueuePdrsSuccessSpec.js [hunk 3]
```diff
@@ -39,6 +46,11 @@ describe('The Discover And Queue PDRs workflow', () => {
     );
   });
 
+  afterAll(async () => {
```
[the remark this answers is not in the document]
Comment (mennovandiermen) [reply example/spec/discoverAndQueuePdrs/DiscoverAndQueuePdrsSuccessSpec.js]: Yeah it kicks off ParsePdr and then (presumably) IngestGranule, I'll check about waiting for those.

File: example/spec/parsePdr/ParsePdrSuccessSpec.js [hunk 4]
```diff
@@ -17,6 +19,14 @@ const taskName = 'ParsePdr';
 
 const expectedParsePdrOutput = JSON.parse(fs.readFileSync('./spec/parsePdr/ParsePdr.output.json'));
 
+const s3data = [
+  '@cumulus/test-data/pdrs/MOD09GQ_1granule_v3.PDR',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf.met',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606_ndvi.jpg',
+  '@cumulus/test-data/granules/L2_HR_PIXC_product_0001-of-4154.h5'
```
[the remark this answers is not in the document]
Comment (mennovandiermen) [reply example/spec/parsePdr/ParsePdrSuccessSpec.js]: These files are referenced in the PDR.

File: example/spec/ingestGranule/IngestGranuleFailureSpec.js [hunk 5]
```diff
@@ -7,14 +7,22 @@ const { aws: { s3 } } = require('@cumulus/common');
...
 
 const granuleRegex = '^MOD09GQ\\.A[\\d]{7}\\.[\\w]{6}\\.006\\.[\\d]{13}$';
 const testDataGranuleId = 'MOD09GQ.A2016358.h13v04.006.2016360104606';
 
+const s3data = [
+  '@cumulus/test-data/pdrs/MOD09GQ_1granule_v3.PDR',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf.met',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606_ndvi.jpg',
+  '@cumulus/test-data/granules/L2_HR_PIXC_product_0001-of-4154.h5'
```
[the remark this answers is not in the document]
Comment (mennovandiermen) [reply example/spec/ingestGranule/IngestGranuleFailureSpec.js]: These files are referenced in the `input.payload.json`

File: example/spec/ingestGranule/IngestGranuleSuccessSpec.js [hunk 6]
```diff
@@ -41,6 +41,14 @@ const templatedOutputPayloadFilename = templateFile({
   config: config[workflowName].IngestGranuleOutput
 });
 
+const s3data = [
+  '@cumulus/test-data/pdrs/MOD09GQ_1granule_v3.PDR',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf.met',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606.hdf',
+  '@cumulus/test-data/granules/MOD09GQ.A2016358.h13v04.006.2016360104606_ndvi.jpg',
+  '@cumulus/test-data/granules/L2_HR_PIXC_product_0001-of-4154.h5'
```
[the remark this answers is not in the document]
Comment (mennovandiermen) [reply example/spec/ingestGranule/IngestGranuleSuccessSpec.js]: These files are referenced in the input.payload.json

**Pull request merged**